modules/serdes/include/hephaestus/serdes/json.h

```diff
@@ -59,16 +64,16 @@ auto serializeToJSON(const T& data) -> std::string {
   }
 }
 
-template <JSONDeserializable T>
-auto deserializeFromJSON(std::string_view buffer, T& data) -> void {
+template <typename T>
+auto deserializeFromJSON(std::string_view json, T& data) -> void {
   if constexpr (protobuf::ProtobufSerializable<T>) {
-    protobuf::deserializeFromJSON(buffer, data);
+    protobuf::deserializeFromJSON(json, data);
   } else if constexpr (HasJSONDeserialization<T>) {
-    fromJSON(buffer, data);
+    fromJSON(json, data);
   } else if constexpr (HasNlohmannJSONSerialization<T>) {
-    data = nlohmann::json::parse(buffer).get<T>();
+    data = nlohmann::json::parse(json).get<T>();
   } else {
-    data = rfl::json::read<T>(buffer).value();
+    data = rfl::json::read<T>(json).value();
```

Comment:
Nit:

This is only taste, but I would prefer template specialization over if/else

Reply:
you mean like 
https://godbolt.org/z/zdhEEzcY1